### PR TITLE
Allow logging in with the account email address

### PR DESCRIPTION
**Pull request by hughsie**
Fixes some of https://github.com/hughsie/lvfs-website/issues/82

### CODE REVIEW

File: app/views.py [hunk 1]
```diff
@@ -214,9 +214,16 @@ def new_account():
 def login():
     """ A login screen to allow access to the LVFS main page """
     # auth check
+    used_deprecated_username = False
     user = db.session.query(User).\
-            filter(User.username == request.form['username']).\
+            filter(User.email == request.form['username']).\
             filter(User.password == _password_hash(request.form['password'])).first()
+    if not user:
```
Comment (superm1) [inline app/views.py]: So what's going to happen if multiple accounts match that email then?  You might need to define that behavior and keep it in test case. Or possibly do a migration in the DB for anything with common email addresses to append a email+user@domain.com for now.

**Comment by superm1** [comment]
The other part is that currently users are able to change their email address in the panel, which effectively changes their login.  Should that stay?  I'm guessing yes - it should only be restricted from changing if you logged in with OAuth.

File: app/views.py (hunk 1 above)
Comment (hughsie) [reply app/views.py]: I was going to just do the `email+user@domain.com` thing manually, and then perhaps make the column unique.

**Comment by hughsie** [comment]
>The other part is that currently users are able to change their email address in the panel, which effectively changes their login. Should that stay? I'm guessing yes

I guessed no -- we'd only let the user change the displayname -- the email address would be the 100% thing that tied the user to the account. If a user changed employment from Dell to Logitech you don't want to "update" the profile on the LVFS, you want a whole new user in the new vendor group.

> - it should only be restricted from changing if you logged in with OAuth.

I think the profile should be "locked" i.e. you can't change your display name either, as this will be coming from OAuth too.

**Comment by hughsie** [comment]
This is too intertwined with the OAuth stuff, I'll close this, merge the branches and open a new PR with it all working. There's no point allowing logins with the email if we don't do OAuth and there's a whole lot of other stuff that makes sense to clear up at the same time.

**Pull request closed without merging**